app/src/main/java/com/android/voyageur/model/notifications/TripInviteRepositoryFirebase.kt

```diff
@@ -0,0 +1,196 @@
+ * @constructor creates a new TripInviteRepositoryFirebase
+ * @property db the firestore instance to use
+ * @property collectionPath the path to the collection of trip invites
+ */
+class TripInviteRepositoryFirebase(private val db: FirebaseFirestore) : TripInviteRepository {
+  private val collectionPath = "friendRequests"
+
+  override fun init(onSuccess: () -> Unit) {
+    db.collection(collectionPath).get().addOnSuccessListener { onSuccess() }
+  }
+
+  /**
+   * Get the number of trip invites for a user
+   *
+   * @param userId the user to get the number of trip invites for
+   * @param onSuccess callback to execute after successful fetch
+   * @param onFailure exception handling callback
+   */
+  override fun getTripInvitesCount(
+      userId: String,
+      onSuccess: (Long) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .whereEqualTo("to", userId)
+        .count()
+        .get(AggregateSource.SERVER)
+        .addOnSuccessListener { onSuccess(it.count) }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  /**
+   * Create a trip invite
+   *
+   * @param req the trip invite to create
+   * @param onSuccess callback to execute after successful creation
+   * @param onFailure exception handling callback
+   */
+  override fun createTripInvite(
+      req: TripInvite,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(req.id)
+        .set(req, SetOptions.merge())
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  /**
+   * Get all trip invites for a user
+   *
+   * @param userId the user to get the trip invites for
+   * @param onSuccess callback to execute after successful fetch
+   * @param onFailure exception handling callback
+   */
+  override fun getTripInvites(
+      userId: String,
+      onSuccess: (List<TripInvite>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .whereEqualTo("to", userId)
+        .get()
+        .addOnSuccessListener { result ->
+          val invites = result.map { document -> document.toObject(TripInvite::class.java) }
+          onSuccess(invites)
+        }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  /**
+   * Delete a trip invite
+   *
+   * @param invite the trip invite to delete
+   * @param onSuccess callback to execute after successful deletion
+   * @param onFailure exception handling callback
+   */
+  override fun deleteTripInvite(
+      invite: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(invite)
+        .delete()
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
```

Comment:
Same here

Reply:
This is not the same. If the document doesn't exist, the onFailure event will be raised